src/r2mm/mods/ProfileModList.ts

```diff
@@ -43,7 +43,7 @@ export default class ProfileModList {
             try {
                 const value = (yaml.parse((await fs.readFile(profile.joinToProfilePath('mods.yml'))).toString()) || []);
                 for(let modIndex in value){
-                    const mod = new ManifestV2().fromReactive(value[modIndex]);
+                    const mod = value[modIndex];
```

Comment:
This could probably be refactored a bit (by just using `value[modIndex]` instead of creating the local `mod` variable, but idk might be that it needs a copy for some reason. Would need testing that it works, so I decided to not use time to do that.)

Reply:
A bit of testing would always be a good idea. In this case it would've caught that the current changes spectacularly break things. A good lesson in "never believe what Antti says, always verify it yourself" 😂 

The issue is that `.fromReactive` was meant for fixing the objects that get passed through a Vue component (e.g. `v-for="(mod) in mods"`). What I failed to notice is that the usage here is completely different - someone has just decided to use the existing method since it has worked for their use case. Here the loop is one where ManifestV2[] has been cast to Yaml, written to disk, read as Yaml from the disk, and parsed into a plain JS object. So here we actually do need to cast it into a ManifestV2 again.

I'm not exactly sure what the best way to solve this would be, as I can think of a few ways. They're outside the scope of the task though, so I'd suggest you restore `ManifestV2.fromReactive` but rename it into something more descriptive like `fromJsObject`.

Reply:
BTW assigning an object into a variable in JS doesn't copy the object, the variable only references the existing object (its location in memory). But yeah, the variable is used here probably for readability reasons and could arguably be removed.